receiver/kubeletstatsreceiver/internal/kubelet/accumulator.go

```diff
@@ -44,71 +43,51 @@ var ValidMetricGroups = map[MetricGroup]bool{
 }
 
 type metricDataAccumulator struct {
-	m                     []pmetric.Metrics
 	metadata              Metadata
 	logger                *zap.Logger
 	metricGroupsToCollect map[MetricGroup]bool
 	time                  time.Time
+	mb                    *metadata.MetricsBuilder
 }
 
-const (
-	scopeName = "otelcol/kubeletstatsreceiver"
-)
-
 func (a *metricDataAccumulator) nodeStats(s stats.NodeStats) {
 	if !a.metricGroupsToCollect[NodeMetricGroup] {
 		return
 	}
 
-	md := pmetric.NewMetrics()
-	rm := md.ResourceMetrics().AppendEmpty()
-	fillNodeResource(rm.Resource(), s)
-
-	ilm := rm.ScopeMetrics().AppendEmpty()
-	ilm.Scope().SetName(scopeName)
-
-	startTime := pcommon.NewTimestampFromTime(s.StartTime.Time)
```

Comment:
We should not drop the start time. It should be provided to the MetricsBuilder factory as an optional argument

Reply:
Good catch.  The start time is coming from the summary data, and the start time for Node is different than the start time for Pod.  Right now the metrics builder is created once, before the summary data is collected.  I think we could set the time using `.Reset` for nodes, but how will we update the time for Pods?

Reply:
Oh different start times per resource is something that is not supported by the metrics builder. This complicates things. I believe we have to have different builders then, and make this metrics merge as was done previously in this PR. Sorry for asking to revert some of that stuff

Reply:
No problem, I can do that tomorrow.